### PR TITLE
feat: allow to configure framework via fluent api

### PR DESCRIPTION
**Pull request by TheNoim**

### CODE REVIEW

File: Connectivity/Classes/Connectivity.swift [hunk 1]
```diff
@@ -309,9 +313,9 @@ private extension Connectivity {
     /// Returns a URL request for an Authorization header if the `bearerToken` property is set,
     /// otherwise `nil` is returned.
     func authorizedURLRequest(with url: URL) -> URLRequest? {
-        guard let bearerToken = self.bearerToken else { return nil }
+        guard let headerValue = bearerToken != nil ? "Bearer \(bearerToken!)" : authorizationHeader != nil ? authorizationHeader : nil  else { return nil }
```
[the remark this answers is not in the document]
Comment (rwbutler) [reply Connectivity/Classes/Connectivity.swift:316]: The two ternaries here are a little hard to follow, I'd be inclined to go with something a little more verbose but readable:

```swift
let authHeader: String? = authorizationHeader
if let bearerToken = bearerToken {
      authHeader = authHeader ?? "Bearer \(bearerToken)"
}
guard let authHeader = authHeader else { 
         return nil 
}
```

File: Connectivity/Classes/Combine/ConnectivitySubscription.swift [hunk 2]
```diff
@@ -17,6 +17,9 @@ class ConnectivitySubscription<S: Subscriber>: Subscription where S.Input == Con
 
     init(configuration: ConnectivityConfiguration, subscriber: S) {
         connectivity = Connectivity(configuration: configuration)
+        connectivity.framework = configuration.framework
+        connectivity.bearerToken = configuration.bearerToken
+        connectivity.authorizationHeader = configuration.authorizationHeader
```
Comment (rwbutler) [inline Connectivity/Classes/Combine/ConnectivitySubscription.swift:22]: There's actually a function in Connectivity.swift called `configure(with:)` which takes the values from the configuration object and applies them - it would sense if these values were applied in the same place.

File: Connectivity/Classes/Connectivity.swift [hunk 3]
```diff
@@ -33,6 +33,10 @@ public class Connectivity: NSObject {
 
     /// Optionally configure a bearer token to be sent as part of an Authorization header.
     public var bearerToken: String?
+    
+    /// Optionally configure the Authorization header manually.
+    /// Note: `bearerToken` has a higher priority
```
Comment (rwbutler) [inline Connectivity/Classes/Connectivity.swift:38]: I've actually changed this around so that the authheader has higher priority - since the bearer token header is just a more specific instance of this I figured this probably has higher utility (see comment below).

**Review by rwbutler — CHANGES_REQUESTED** [review]
@TheNoim thanks for the PR! 🙏 Appreciate the contribution - thanks for spotting that certain values weren't configurable via the configuration object. I can the value in being able to set an auth header so this change makes sense to me - I've added a couple of small comments which should be easy to address.

**Review by rwbutler — APPROVED** [review]
Thanks for the contribution! Have merged 👍

File: Connectivity/Classes/Connectivity.swift [hunk 4]
```diff
@@ -33,6 +33,10 @@ public class Connectivity: NSObject {
 
     /// Optionally configure a bearer token to be sent as part of an Authorization header.
     public var bearerToken: String?
+
```
[the remark this answers is not in the document]
Comment (y7aa) [reply Connectivity/Classes/Connectivity.swift:36]: > Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
